lib/toolbox/Taskfile.yaml

```diff
@@ -1,59 +1,96 @@
   VERSION:
     sh: git rev-parse --short HEAD
-  REGISTRY_URL_FULL: "{{.REGISTRY_URL}}/{{.LATEST_IMAGE_TAG}}"
-  CONTAINERFILE_PATH: '{{.CONTAINERFILE_PATH | default "./Dockerfile"}}'
+  TOOLBOX_NAME: '{{.TOOLBOX_NAME | default "toolbox"}}'
+  LATEST_IMAGE_TAG: "{{.TOOLBOX_NAME}}:latest"
+  VERSION_IMAGE_TAG: "{{.TOOLBOX_NAME}}:{{.VERSION}}"
+  REGISTRY_LATEST_URL: "{{.REGISTRY_URL_BASE}}/{{.LATEST_IMAGE_TAG}}"
+  REGISTRY_VERSION_URL: "{{.REGISTRY_URL_BASE}}/{{.VERSION_IMAGE_TAG}}"
+  CONTAINERFILE_PATH: '{{.CONTAINERFILE_PATH | default "./Containerfile"}}'
   AWS_REGION: '{{.AWS_REGION | default "us-east-1"}}'
 
 tasks:
-  all:
+  auth:
+    desc: Authenticate to AWS ECR
+    internal: true
+    requires:
+      vars:
+        - AWS_REGION
+        - REGISTRY_URL_BASE
+    cmds:
+      - aws ecr get-login-password --region {{.AWS_REGION}} | docker login --username AWS --password-stdin {{.REGISTRY_URL_BASE}}
+
+  install:
+    desc: Install wrapper script from geodesic container
+    internal: true
+    requires:
+      vars:
+        - TOOLBOX_NAME
+    cmds:
+      - docker run --rm {{.REGISTRY_LATEST_URL}} init | bash -s {{.TOOLBOX_NAME}}
+
+  run:build:
     desc: Build, install, and run the toolbox image
     cmds:
       - task: build
-      - task: install
       - task: run
 
+  run:pull:
+    desc: Pull, install, and run the toolbox image
+    cmds:
+      - task: pull
+      - task: run
+
+  clean:
+    desc: Cleans all toolbox images off of the local machine
+    prompt: Are you sure you want to clean all {{.TOOLBOX_NAME}} toolbox images off of the local machine?
+    cmds:
+      - docker rmi $(docker images --format '{{ "{{" }}.Repository{{ "}}" }}:{{ "{{" }}.Tag{{ "}}" }}' | grep '{{.TOOLBOX_NAME}}')
+
   build:
     desc: Build and tag our toolbox image
     requires:
       vars:
-        - IMAGE_NAME
+        - TOOLBOX_NAME
         - CONTAINERFILE_PATH
     cmds:
-      - "docker build --tag {{.LATEST_IMAGE_TAG}} --tag {{.VERSION_IMAGE_TAG}} -f {{.CONTAINERFILE_PATH}} ."
+      - |
+        docker build --tag {{.LATEST_IMAGE_TAG}} \
+                     --tag {{.VERSION_IMAGE_TAG}} \
+                     --tag {{.REGISTRY_LATEST_URL}} \
+                     --tag {{.REGISTRY_VERSION_URL}} \
+                     -f {{.CONTAINERFILE_PATH}} .
       - echo "Done building and tagging toolbox 💯 Tagged as {{.VERSION_IMAGE_TAG}} + {{.LATEST_IMAGE_TAG}}"
 
-  install:
-    desc: Install wrapper script from geodesic container
-    requires:
-      vars:
-        - IMAGE_NAME
+  run:
+    desc: Run our toolbox image while also mounting your $HOME folder to `/localhost` in the container
+    deps:
+      - install
     cmds:
-      - docker run --rm {{.LATEST_IMAGE_TAG}} init | bash -s {{.IMAGE_NAME}} || (echo 'Try "sudo make install"'; exit 1)
+      - "{{.TOOLBOX_NAME}}"
 
   pull:
     desc: Pull our toolbox image from registry
+    deps:
+      - auth
     requires:
       vars:
         - REGISTRY_URL_BASE
```

Comment:
Why is it required?

Reply:
@gberenice `REGISTRY_URL_BASE` does not have a default and without a value, the `REGISTRY_*` vars will be broken. I added requiring here to ensure that we're passing a value from `.env.taskit`. Maybe there is a better way to do this?

Reply:
Hm, probably not. I don't see a way to add a check to the global `vars` section. LATEST_IMAGE_TAG has a default, so probably just REGISTRY_URL_BASE needs to be required. But I don't feel strong about this, and I get why we'd want to check them all.